pyfar/classes/coordinates.py

```diff
@@ -1487,74 +1978,40 @@ def _set_points(self, points_1, points_2, points_3,
         [L,M,...,N, 3].
         """
 
-        # get coordinate system against the points are checked
-        if system is None:
-            system = self._system
-
         # cast to numpy array
-        pts = [np.atleast_2d(np.asarray(points_1, dtype=np.float64)),
-               np.atleast_2d(np.asarray(points_2, dtype=np.float64)),
-               np.atleast_2d(np.asarray(points_3, dtype=np.float64))]
+        x = np.atleast_1d(np.asarray(x, dtype=np.float64))
+        y = np.atleast_1d(np.asarray(y, dtype=np.float64))
+        z = np.atleast_1d(np.asarray(z, dtype=np.float64))
 
         # transpose
-        for nn, p in enumerate(pts):
-            if len(p.shape) == 2 and p.shape[0] == 1:
-                pts[nn] = np.transpose(p)
+        if len(x.shape) == 2 and (x.shape[0] == 1 or x.shape[1] == 1):
+            x = x.flatten()
+        if len(y.shape) == 2 and (y.shape[0] == 1 or y.shape[1] == 1):
+            y = y.flatten()
+        if len(z.shape) == 2 and (z.shape[0] == 1 or z.shape[1] == 1):
+            z = z.flatten()
```

Comment:
I'm not sure what this should do, but it looks like you could use np.squeeze? Why does it say transpose?

Reply:
yes right, squeeze is better

Reply:
the difference between this implementation and squeeze is:
previously:
(1,2) -> (2,)
(2,1) -> (2,)
(3,2,1) -> (3,2,1)
with squeeze:
(3,2,1)  -> (3,2)
so the question is, what do we want? maybe we should let the user decide and we shouldn't do any squeeze.